catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraProduction.scala

```diff
@@ -0,0 +1,169 @@
+    varFields.map { vf =>
+      val places = placesFromSubfields(vf, subfieldTag = "a")
+      val agents = agentsFromSubfields(vf, subfieldTag = "b")
+      val dates = datesFromSubfields(vf, subfieldTag = "c")
+
+      val extraPlaces = placesFromSubfields(vf, subfieldTag = "e")
+      val extraAgents = agentsFromSubfields(vf, subfieldTag = "f")
+      val extraDates = datesFromSubfields(vf, subfieldTag = "g")
+
+      val productionFunction =
+        if (extraPlaces != Nil || extraAgents != Nil || extraDates != Nil) {
+          Some(Concept(label = "Manufacture"))
+        } else None
+
+      ProductionEvent(
+        places = places ++ extraPlaces,
+        agents = agents ++ extraAgents,
+        dates = dates ++ extraDates,
+        function = productionFunction
+      )
+    }
+
+  // Populate wwork:production from MARC tag 264.
+  //
+  // The rules are as follows:
+  //
+  //  - Populate "places" from subfield "a" and type as "Place"
+  //  - Populate "agents" from subfield "b" and type as "Agent"
+  //  - Populate "dates" from subfield "c" and type as "Period"
+  //
+  // The production function is set based on the second indicator, as defined
+  // in the MARC spec.
+  //
+  //  - 0 = Production
+  //  - 1 = Publication
+  //  - 2 = Distribution
+  //  - 3 = Manufacture
+  //
+  // Note that a, b and c are repeatable fields.
+  //
+  // https://www.loc.gov/marc/bibliographic/bd264.html
+  //
+  private def getProductionFrom264Fields(varFields: List[VarField]) =
+    varFields.map { vf =>
+      val places = placesFromSubfields(vf, subfieldTag = "a")
+      val agents = agentsFromSubfields(vf, subfieldTag = "b")
+      val dates = datesFromSubfields(vf, subfieldTag = "c")
+
+      val productionFunctionLabel = vf.indicator2 match {
+        case Some("0") => "Production"
+        case Some("1") => "Publication"
+        case Some("2") => "Distribution"
+        case Some("3") => "Manufacture"
+        case other =>
+          throw GracefulFailureException(new RuntimeException(
+            s"Unrecognised second indicator for production function: [$other]"
+          ))
+      }
+
+      val productionFunction = Some(Concept(label = productionFunctionLabel))
+
+      ProductionEvent(
+        places = places,
+        agents = agents,
+        dates = dates,
+        function = productionFunction
+      )
+    }
+
+  private def placesFromSubfields(vf: VarField,
+                                  subfieldTag: String): List[Place] =
+    vf.subfields
+      .filter { _.tag == subfieldTag }
+      .map { sf: MarcSubfield =>
+        Place(label = sf.content)
+      }
+
+  private def agentsFromSubfields(
+    vf: VarField,
+    subfieldTag: String): List[Unidentifiable[Agent]] =
+    vf.subfields
+      .filter { _.tag == subfieldTag }
+      .map { sf: MarcSubfield =>
+        Agent(label = sf.content)
+      }
+      .map { ag: Agent =>
+        Unidentifiable(ag)
+      }
+
```

Comment:
can this be a single map to `Unidentifiable(Agent(label = sf.content))`?

Reply:
I thought I’d tried that and it didn’t work, but it seems happy enough now. Changed!